flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -877,6 +867,18 @@ public ApplicationReport startAppMaster(
 		Path remoteYarnSiteXmlPath = null;
 		boolean hasKrb5 = false;
 		if (System.getenv("IN_TESTS") != null) {
+			File f = new File(System.getenv("YARN_CONF_DIR"), Utils.YARN_SITE_FILE_NAME);
```

Comment:
Are we sure that this file always exists? What happens if this file does not exist? I think it would be best if we keep the test code as close to the production code as possible. In this case it would mean that we set the environment of the Yarn container up in a way that we don't have to explicitly ship the `yarn-site.xml` to it, if possible.